### PR TITLE
Fix gotype partition strategy

### LINKED ISSUE #348: go type: passing the wrong type of arguments
I'm executing gometalinter via
```
gometalinter ./src
```
And getting the following in the debug output: 
```
executing /Users/charlesrudolph/go/bin/gotype -e ./src/cmd ./src/drivers ./src/exit ./src/git ./src/prompt ./src/script ./src/steps ./src/util
DEBUG: [gotype.9]: warning: /Users/charlesrudolph/go/bin/gotype returned exit status 2: read src/cmd: is a directory
```
It appears when passing multiple arguments to go type it expects each to be a file. To pass directories, I believe we need separate executions.

### PR DESCRIPTION
**Pull request by dnephin**
Fixes #348 
Branched from #350 (only the last commit is new)

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -278,10 +278,10 @@ func outputToConsole(issues chan *Issue) int {
 	return status
 }
 
-func outputToJSON(issues chan *Issue) int {
+func outputToJSON(chIssues chan *Issue) int {
```
Comment (alecthomas) [inline main.go]: What's the purpose of all these issues -> chIssues renames?

Comment (dnephin) [reply main.go]: When `issues` was a separate package the variable name was masking the package name, so i had to change the variable. Now that the package is gone I could revert these.

File: issue.go [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+	Line       int      `json:"line"`
+	Col        int      `json:"col"`
+	Message    string   `json:"message"`
+	formatTmpl *template.Template
+}
+
+// NewIssue returns a new issue. Returns an error if formatTmpl is not a valid
+// template for an Issue.
+func NewIssue(linter string, formatTmpl *template.Template) (*Issue, error) {
+	if formatTmpl == nil {
```
[the remark this answers is not in the document]
Comment (dnephin) [reply issue.go]: Ah good call. Previously there was when this type was being used by regresiontests, but not anymore.  Removed

File: main.go (hunk 1 above)
Comment (dnephin) [reply main.go]: Reverted these changes

**Pull request merged**